### PR TITLE
[Mage] Add Soul of the Archmage to items

### PR DESCRIPTION
**Pull request by avilene**
This is done so the player can see their legendaries; based off the Highlord one from paladin,

### CODE REVIEW

File: src/Parser/Mage/Fire/Modules/Items/SoulOfTheArchmage.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import Combatants from 'Parser/Core/Modules/Combatants';
+
+class SoulOfTheArchmage extends Module {
+  static dependencies = {
+    combatants: Combatants,
+  };
+
+  on_initialized() {
+    this.active = this.combatants.selected.hasFinger(ITEMS.SOUL_OF_THE_ARCHMAGE.id);
+    console.log(this.combatants.selected.hasFinger(ITEMS.SOUL_OF_THE_ARCHMAGE.id));
```
Comment (sMteX) [inline src/Parser/Mage/Fire/Modules/Items/SoulOfTheArchmage.js]: The log here is unnecessary (or prepend it with `debug &&`).

**Review by sMteX — CHANGES_REQUESTED** [review]
Just the tiny log in Fire one, otherwise looks good. Thanks for the PR :)

**Comment by kfinch** [comment]
These are fine as placeholders, but I may end up overwriting the Frost one when I implement Frozen Touch

File: src/Parser/Mage/Fire/Modules/Items/SoulOfTheArchmage.js (hunk 1 above)
Comment (fyruna) [reply src/Parser/Mage/Fire/Modules/Items/SoulOfTheArchmage.js]: Removed in 1c149be

**Comment by fyruna** [comment]
@kfinch Obviously, but we should always make sure the player knows which legendaries they have; I would like to keep this around unless your Frozen Touch addition will actually include showing that the player has this equipped.

**Review by sMteX — APPROVED** [review]

**Pull request merged**